File: problems/648-replace-words/replace-words.js
/**
 * @param {string[]} dictionary
 * @param {string} sentence
 * @return {string}
 */
const replaceWords = (dictionary, sentence) => {
    let arraySentence = sentence.split(" ");
    for (let i = 0; i < arraySentence.length; i++) {
        for (let item of dictionary) {
            if (arraySentence[i].startsWith(item)) {
                arraySentence[i] = item;
            }
        }
    }
    return arraySentence.join(" ")
};

File: problems/648-replace-words/replace-words.py
class Solution:
    def replaceWords(self, dictionary: List[str], sentence: str) -> str:
        sentenceList = sentence.split() 
        for i in range(len(sentenceList)):
            for item in dictionary:
                if sentenceList[i].startswith(item):
                    sentenceList[i] = item
        
        return " ".join(sentenceList)